scio-core/src/main/java/com/spotify/scio/util/RemoteFileUtil.java

```diff
@@ -0,0 +1,320 @@
+/*
```

Comment:
It's mostly using Java APIs. And this class maybe reused in a pure Java DF pipeline.